packages/paste-core/components/radio-button-group/src/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+//import type {BoxProps} from '@twilio-paste/box';
+import type {InlineControlGroupProps} from '@twilio-paste/inline-control-group';
+import {HelpText} from '@twilio-paste/help-text';
+import {Label} from '@twilio-paste/label';
+
+import {RadioButtonContext} from './RadioButtonContext';
+import {AttachedRadioButtonGroupWrapper, UnattachedRadioButtonGroupWrapper} from './StyledRadioButtonGroupWrapper';
+
+/*
+NOTE: not sure why this was added originally. seems we use the import on line 5 above?
+GA: this interface  uses ref?: HTMLFieldSetElement as opposed to ref?: any like in the imported InlineControlGroupProps
+Could maybe just extend it more concisely in RadioButtonGroupProps?
+
+interface InlineControlGroupProps
+  extends Pick<BoxProps, 'element'>,
+    Omit<React.FieldsetHTMLAttributes<HTMLFieldSetElement>, 'onChange'> {
+  children: React.ReactNode;
+  disabled?: boolean;
+  errorText?: string | React.ReactNode;
+  helpText?: string | React.ReactNode;
+  legend: string | NonNullable<React.ReactNode>;
+  orientation?: 'vertical' | 'horizontal';
+  ref?: HTMLFieldSetElement;
+  required?: boolean;
+  i18nRequiredLabel?: string;
+}
+*/
+
+export interface RadioButtonGroupProps extends InlineControlGroupProps {
+  name: string;
+  onChange?: (value: string) => void; // the context has this as taking in an event, is that ok?
+  value?: string;
+  i18nRequiredLabel?: string;
+  attached?: boolean;
+  // ref?: HTMLFieldSetElement;
+}
+
+const RadioButtonGroup = React.forwardRef<HTMLFieldSetElement, RadioButtonGroupProps>(
+  (
+    {
+      name,
+      element = 'RADIO_BUTTON_GROUP',
+      value,
+      onChange,
+      disabled = false,
+      errorText,
+      children,
+      i18nRequiredLabel = '(required)',
+      helpText,
+      legend,
+      required,
+      attached = false,
+      ...props
+    },
+    ref
+  ) => {
+    const onChangeHandler = React.useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>): void => {
+        if (onChange != null) {
+          onChange(event.target.value);
+        }
+      },
+      [onChange]
+    );
+
+    const contextValue = React.useMemo(() => {
+      return {
+        name,
+        value: value || '',
+        disabled,
+        hasError: errorText != null,
+        onChange: onChangeHandler,
+      };
+    }, [name, value, disabled, errorText, onChangeHandler]);
+
+    return (
+      // Do we need to spread this contextValue object here?
+      <RadioButtonContext.Provider value={contextValue}>
```

Comment:
Not necessarily... the difference between this: `value={contextValue}` and `value={{..contextValue}}` is that `value={{..contextValue}}` makes a new object with the same key/values as contextValue. This is copy pasta from RadioGroup so I think its fine.